### DISCUSSION TITLE
Sometimes the code run too fast and sometimes my website doesnt finish to refresh itself therefore the code failes

### DISCUSSION THREAD

**Discussion by elnatan2604** [seleniumbase/SeleniumBase#2153] [Q&A] [answered]
I have a code for example: 

self.send_keys(fn.FieldsNames.id_html, final_id_pick)
self.wait_for_element_clickable(fn.FieldsNames.continue_button2_id, timeout=60)
self.click(fn.FieldsNames.continue_button2_id)

and my website when it puts the id in his box, it needs to "refresh" itself and update the id box accordingly, this takes something between to 1 second to 5 seconds.
I put the wait_for_element_clickable but it still sometimes doesnt work and the code fails.
So my question is if there is any other wait that I can use to do it? is there a wait thats waits for the website to finish "refresh" itself and then continues?

**Comment by mdmintz** [comment, marked as answer]
`self.wait_for_element_clickable()` waits for `element.is_displayed()` and `element.is_enabled()` (from the raw selenium built-in methods). It's not necessarily enough, depending on what you're actually clicking on, for the website you're using. You may need to wait for another element to be visible (for example) or something else. (Eg. `self.wait_for_element_visible(selector)`)

When you're not sure what else you need to wait on, that's when you may be forced to use `time.sleep(seconds)` / `self.sleep(seconds)` in the meantime until you know exactly what you need to wait for.
